### PR TITLE
OGS compatibility

### PR DESCRIPTION
**Pull request by jbathmann**
This PR is giving functionality to use pyMANGA in python boundary conditions of OGS
It includes

- Class to allow for external modification of project
- New timestepper
- External accessibility

@jvollhueter could you please have a look what updates are required in you testscripts as main.py has been renamed...?
@mcwimm I think we need a simple bg-competition concept to test. Maybe we could start with the simple-test case?

### CODE REVIEW

File: TimeLoopLib/ExternalDynamicTimeStepping.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@date: 2018-Today
+@author: jasper.bathmann@ufz.de
+"""
+
```
Comment (mcwimm) [inline TimeLoopLib/ExternalDynamicTimeStepping.py:7]:
```suggestion
from TimeLoopLib.TreeDynamicTimeStepping import TreeDynamicTimeStepping


class ExternalDynamicTimeStepping(TreeDynamicTimeStepping):
    def __init__(self, project, t_0):
        super().__init__(project)
        ## Usability for OGS & check if BG concept works with OGS
        self.t_step_begin = t_0

        #TODO: We need a nice check for ogs usability?!
        try:
            self.belowground_competition.getOGSAccessible()
        except AttributeError:
            raise AttributeError(
                """In order to use MANGA as OGS python boundary condition,
                    one has to use a corresponding belowground competition
                    concept. Please see documentation for further details!""")

    def step(self, t_end):
        t_start = self.t_step_begin
        super().step(t_start, t_end)
        self.t_step_begin = t_end
```

File: TimeLoopLib/ExternalDynamicTimeStepping.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        self.aboveground_competition = project.getAbovegroundCompetition()
+        self.belowground_competition = project.getBelowgroundCompetition()
+        self.death_and_growth_concept = project.getDeathAndGrowthConcept()
+        self.population = project.getPopulation()
+        ## Output configuration
+        self.tree_output = project.getTreeOutput()
+        ## Usability for OGS & check if BG concept works with OGS
+        self.t_step_begin = t_0
+
+        #TODO: We need a nice check for ogs usability?!
```
Comment (mcwimm) [inline TimeLoopLib/ExternalDynamicTimeStepping.py:20]: What did you have in mind?

File: TreeModelLib/BelowgroundCompetition/SimpleTest/SimpleTest.py [hunk 3]
```diff
@@ -43,3 +43,18 @@ def prepareNextTimeStep(self, t_ini, t_end):
...
         self.trees.append(1)
+
+    ## Test for external timestepper, whether the concept is optimized for ex-
+    #  ternal communication
+    def getOGSAccessible(self):
+        return True
+
+    ## Setter for external information - specify and document argument for each
+    #  concept application
+    #  (See XMLtoProject.py as example)
```
Comment (mcwimm) [inline TreeModelLib/BelowgroundCompetition/SimpleTest/SimpleTest.py:54]: this is not an useful example, imo

File: TreeOutputLib/OneTreeOneFile/OneTreeOneFile.py [hunk 4]
```diff
@@ -13,16 +13,16 @@
 #  A line containing time, position, desired geometric measures and desired
 #  parameters is written at every nth timestep.
 class OneTreeOneFile(OneTimestepOneFile):
-    
-## Constructor of dummy objects in order to drop output
+
+    ## Constructor of dummy objects in order to drop output
     #  @param args xml element parsed from project to this constructor.
     def __init__(self, args):
         OneTimestepOneFile.__init__(self, args)
```
Comment (mcwimm) [inline TreeOutputLib/OneTreeOneFile/OneTreeOneFile.py:20]:
```suggestion
        super().__init__(self, args)
```

**Review by mcwimm — COMMENTED** [review]
Good start! I like it :)

File: TimeLoopLib/ExternalDynamicTimeStepping.py (hunk 2 above)
Comment (jbathmann) [reply TimeLoopLib/ExternalDynamicTimeStepping.py:20]: Just an attribute to be checked. "...self.can_be_used_together_with_ogs_and_is_therefore_supercool" or similar...

File: TimeLoopLib/ExternalDynamicTimeStepping.py (hunk 1 above)
Comment (jbathmann) [reply TimeLoopLib/ExternalDynamicTimeStepping.py:7]: changed. Thanks!

File: TreeModelLib/BelowgroundCompetition/SimpleTest/SimpleTest.py (hunk 3 above)
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/SimpleTest/SimpleTest.py:54]: updated

File: TreeOutputLib/OneTreeOneFile/OneTreeOneFile.py (hunk 4 above)
Comment (jbathmann) [reply TreeOutputLib/OneTreeOneFile/OneTreeOneFile.py:20]: the changes here are only related to yapf. Hence, I'd keep this change for the next pr related to output? WDYT? #107

Comment (mcwimm) [reply TreeOutputLib/OneTreeOneFile/OneTreeOneFile.py:20]: Yes, that's fine.

File: TimeLoopLib/ExternalDynamicTimeStepping.py (hunk 2 above)
Comment (mcwimm) [reply TimeLoopLib/ExternalDynamicTimeStepping.py:16]: Ok. But isn't this already done with the `self.getOGSAccessible = True` statement in each concept?

**Pull request merged**